apps/prairielearn/src/lib/ai.ts

```diff
@@ -59,32 +62,25 @@ export function calculateResponseCost({
   return cachedInputCost + inputCost + outputCost;
 }
 
-export function emptyUsage(): OpenAI.Responses.ResponseUsage {
+export function emptyUsage(): LanguageModelUsage {
   return {
-    input_tokens: 0,
-    input_tokens_details: { cached_tokens: 0 },
-    output_tokens: 0,
-    output_tokens_details: { reasoning_tokens: 0 },
-    total_tokens: 0,
+    inputTokens: 0,
+    cachedInputTokens: 0,
+    outputTokens: 0,
+    reasoningTokens: 0,
+    totalTokens: 0,
   };
 }
 
 export function mergeUsage(
-  a: OpenAI.Responses.ResponseUsage | undefined,
-  b: OpenAI.Responses.ResponseUsage | undefined,
-): OpenAI.Responses.ResponseUsage {
+  a: LanguageModelUsage | undefined,
+  b: LanguageModelUsage | undefined,
+): LanguageModelUsage {
   return {
-    input_tokens: (a?.input_tokens ?? 0) + (b?.input_tokens ?? 0),
-    input_tokens_details: {
-      cached_tokens:
-        (a?.input_tokens_details.cached_tokens ?? 0) + (b?.input_tokens_details.cached_tokens ?? 0),
-    },
-    output_tokens: (a?.output_tokens ?? 0) + (b?.output_tokens ?? 0),
-    output_tokens_details: {
-      reasoning_tokens:
-        (a?.output_tokens_details.reasoning_tokens ?? 0) +
-        (b?.output_tokens_details.reasoning_tokens ?? 0),
-    },
-    total_tokens: (a?.total_tokens ?? 0) + (b?.total_tokens ?? 0),
+    inputTokens: (a?.inputTokens ?? 0) + (b?.inputTokens ?? 0),
```

Comment:
That same util `ZeroedLanguageModelUsage` could be used here to simplify the nulls in this code.

Reply:
See https://github.com/PrairieLearn/PrairieLearn/pull/13121#discussion_r2440416741.